src/log/spec.cr

```diff
@@ -0,0 +1,166 @@
+require "spec/dsl"
+
+class Log
+  # Returns and yields an `EntriesChecker` that allows checking specific log entries
+  # were emitted.
+  #
+  # This capture will even work if there are currently no backends configured, effectively
+  # adding a momentarily backend.
+  #
+  # ```
+  # require "spec"
+  # require "log"
+  # require "log/spec"
+  #
+  # Log.setup(:none)
+  #
+  # def greet(name)
+  #   Log.info { "Greeting #{name}" }
+  # end
+  #
+  # it "greets" do
+  #   Log.capture do |logs|
+  #     greet("Harry")
+  #     greet("Hermione")
+  #     greet("Ron")
+  #
+  #     logs.check(:info, /greeting harry/i)
+  #     logs.next(:info, /greeting hermione/i)
+  #   end
+  # end
+  # ```
+  #
+  # By default logs of all sources and with severity `Severity::Info` or above will be captured.
+  #
+  # Use *level* to only capture of the given severity or above.
+  #
+  # Use *source* to narrow which source are captured. Values that represent single pattern like `http.*` are allowed.
+  #
+  # The `EntriesChecker` will hold a list of emitted entries.
+  #
+  # `EntriesChecker#check`  will look for the first entry that matches the level and message.
+  # `EntriesChecker#next` will validate if the following entry in the list matches the given level and message.
+  # `EntriesChecker#clear` will clear the emitted and captured entries.
+  #
+  # With these methods it is possible to express expected traces in a strict and loosely way, yet always ordered.
+  #
+  # `EntriesChecker#entry` returns the last matched `Entry`. Useful to check additional entry properties other than the message.
+  #
+  # `EntriesChecker#empty` validates there are no pending entries to match.
+  #
+  # Using the yielded `EntriesChecker` allows clearing the entries between statements.
+  #
+  # Invocations can be nested in order to capture each source in their own `EntriesChecker`.
+  #
+  def self.capture(source : String = "*", level : Severity = :info, *, builder = Log.builder)
```

Comment:
Can we capture all the logs by default? (`:trace`)

Reply:
I wanted to preserve the same default as in `Log.setup`, but I was doubting whether to use `:info` or `:trace` as a default.